packages/front-end/components/Search/SearchFilters.tsx

```diff
@@ -0,0 +1,349 @@
+import React, {
+  ChangeEvent,
+  FC,
+  Fragment,
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { Box, Flex, Heading, IconButton } from "@radix-ui/themes";
+import { FaAngleDown, FaAngleUp, FaCheck } from "react-icons/fa";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import {
+  DropdownMenu,
+  DropdownMenuItem,
+} from "@/components/Radix/DropdownMenu";
+import { SearchTermFilterOperator, SyntaxFilter } from "@/services/search";
+import Field from "@/components/Forms/Field";
+
+// Common interfaces
+export interface SearchFiltersItem {
+  id: string;
+  name: string | JSX.Element;
+  searchValue: string;
+  operator?: SearchTermFilterOperator;
+  negated?: boolean;
+  filter?: string;
+  hr?: boolean;
+  disabled?: boolean;
+}
+
+export interface BaseSearchFiltersProps {
+  searchInputProps: {
+    value: string;
+    onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  };
+  syntaxFilters: SyntaxFilter[];
+  setSearchValue: (value: string) => void;
+}
+
+export const FilterHeading: FC<{
+  heading: string;
+  open: boolean;
+}> = ({ heading, open }) => {
+  return (
+    <IconButton
+      variant="ghost"
+      color="gray"
+      radius="small"
+      size="3"
+      highContrast
+    >
+      <Flex gap="2" align="center">
+        <Flex gap="0" align="center">
+          {heading}
+        </Flex>
+        {open ? <FaAngleUp /> : <FaAngleDown />}
+      </Flex>
+    </IconButton>
+  );
+};
+
+export const FilterItem: FC<{
+  item: string | JSX.Element;
+  exists: boolean;
+}> = ({ item, exists }) => {
+  return (
+    <Box className="position-relative">
+      {exists ? (
+        <Box
+          className="position-absolute"
+          style={{ left: "-2px", fontSize: "0.8rem" }}
+        >
+          <FaCheck />{" "}
+        </Box>
+      ) : (
+        ""
+      )}
+      <Box pl="4">{item}</Box>
+    </Box>
+  );
+};
+
+export const FilterDropdown: FC<{
+  filter: string;
+  items: SearchFiltersItem[];
+  syntaxFilters: SyntaxFilter[];
+  open: string;
+  setOpen: (value: string) => void;
+  updateQuery: (filter: SyntaxFilter) => void;
+  operator?: string;
+  heading?: string;
+}> = ({
+  filter,
+  items,
+  open,
+  setOpen,
+  updateQuery,
+  syntaxFilters,
+  operator = "",
+  heading,
+}) => {
+  const [filterSearch, setFilterSearch] = useState<string>("");
+  const showSearchFilter = useMemo(() => items.length > 10, [items]);
```

Comment:
Radix dropdowns have built-in type to search.  It only does a prefix search, so not quite as flexible as yours, but I think the UX is much better with the Radix one - the keyboard navigation feels much more natural.  Would simplify the code a lot here too.